pkg/agent/cniserver/ipam/antrea_ipam.go

```diff
@@ -152,23 +152,25 @@ func (d *AntreaIPAM) Del(args *invoke.Args, k8sArgs *argtypes.K8sArgs, networkCo
 		return err
 	}
 
-	mine, allocator, _, _, err := d.owns(k8sArgs)
+	// When AntreaIPAM.owns() is called for a NodeIPAM Pod, the IPPool cannot be found for the Pod. We pass
+	// swallowNotFoundError to the call to sallow the "NotFound" error.
+	mine, allocator, _, _, err := d.owns(k8sArgs, swallowNotFoundError)
```

Comment:
I'm not sure why it should be fixed in this way. I have two questions:
1. When a Pod is deleted, isn't CNI DEL called before the Pod is removed from the apiserver normally? I can imagine in some cases it could happen that the Pod is gone when it handles CNI DEL like Pod's delete grace period is set to 0, but is it your case?
2. Should IPPool NotFound and Pod NotFound be handled differently? If Pod is not found, I think we are not sure whether this is owned by AntreaIPAM and should return "mine" as false. If Pod is found but IPPool is not, it should at least log error (as we don't allow deleting IPPool in use so this is not expected). But anyway this is for a AntreaIPAM pod, not like the comment explains?

Reply:
1. From e2e test, we can confirm that Pod may not exist when CNI DEL called. Thus we need to handle this case.
2. Nice catch. I think it is harmless if we process `IPPoolNotFound` as current implementation which returns mine=false. Do you think we should check if `ErrStatus.Details.Kind` is `Pod`?

Reply:
For 2, it will invoke an unnecessary external call and hides a mistake which might in the code base.

I think the problem is the current return value from `own` is not clear, its callers have to inject many code for their own business logic. For example:
When it's called by `Add`, it's expected to return `mine` as `true` when the Pod is not found so `Add` can fail immediately, but the actual logic is:
When the Pod is not found, we don't know which IPAM owns it, and its creation should be defered until the Pod is received, or if the Pod has been deleted from API, we won't receive more request anyway.

When it's called by `Del`, it's expected to return `mine` as `false` when the Pod is not found so `Del` can call other IPAM plugins to ensure it's cleaned up completely.

It's obscure to understand the logic by injecting them to the `own` method and why `own` returns different value in different context for same situation.

To put the business logic back to the caller, I think `own` should be more clear what it returns:
If it should only return `mine` as `true` or `false` when it really knows, otherwise return `unknown`. `Add` should call next plugin only for `false`, `Del` should call next plugin for `unknown` and `false`. 
In this way, we don't mix PoolNotFound and PodNotFound:
For `Add`, if the Pod is found but Pool is not found, `own` should return `mine` as `true` and an error to indicate PoolNotFound, the request failed immediately with clear message about PoolNotFound.
For `Add`, if the Pod is not found, `own` should return `unknown` and an error to indicate PodNotFound, the request failed immediately with clear message about PodNotFound.
For `Del`, if the Pod is found but Pool is not found, `own` should return `mine` as `true` and an error to indicate PoolNotFound, the request terminated and an error should be logged for this unexpected situation.
For `Del`, if the Pod is not found, `own` should return `unknown` and an error to indicate PodNotFound, `Del` should call the next plugin until one returns `true`.

Reply:
Thanks. Refactored this part and added comments.